03-Gissell-Bustos/wk02/1-mon/mta-1.js

```diff
@@ -0,0 +1,90 @@
+const subway = {
+    N:['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
+    L:['8th', '6th', 'Union Square', '3rd', '1st'],
+   '6':['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
+   }
+   let ruotes = [];
+    
+    function sameLine(lineOn, stationOn, lineOff, stationOff){
+    
+          let stationOnIndex = subway[lineOn].indexOf(stationOn);
+          let stationOffIndex = subway[lineOff].indexOf(stationOff);
+    
+          if(stationOnIndex > stationOffIndex){
+           
+            for (let index = stationOnIndex; index >= stationOffIndex; index--){
+              ruotes.push({
+                line:lineOn,
+                station: subway[lineOn][index]
+              });
+            }
+          }else{
+            for(let index = stationOnIndex; index <= stationOffIndex; index++){
+              ruotes.push({
+                line:lineOn,
+                station: subway[lineOn][index]
+              });
+        
+            }
+          }
+       
+        ruotes;
```

Comment:
```suggestion
```
I think you meant to `return` this value, although also note that your function doesn't need to `return` anything since you've used `.push()` to add values into the `routes` array.

Just a reminder that using `.push()` is a "side-effect" which makes your function "impure". That's not necessarily an issue but if you wanted to keep your function pure then you would need to create a **new array** and `return` it. A benefit of this is that you could easily separate your routes into two arrays when you call your functions, one for the stops you travel through before you change lines and one for after.
e.g.
```js
const routesBeforeChange = differentLineOn('N', 'Times Square');
const routesAfterChange = differentLineOff('L', '3rd');
```